conmon-rs/server/src/child_reaper.rs

```diff
@@ -104,12 +105,12 @@ impl ChildReaper {
         Ok(grandchild_pid)
     }
 
-    pub fn watch_grandchild(&self, child: Child) -> Result<Receiver<i32>> {
+    pub fn watch_grandchild(&self, child: Child) -> Result<Receiver<ExitChannelData>> {
         let locked_grandchildren = Arc::clone(&self.grandchildren);
         let mut map = lock!(locked_grandchildren);
         let reapable_grandchild = ReapableChild::from_child(&child);
 
-        let (exit_tx, exit_rx) = reapable_grandchild.watch();
+        let (exit_tx, exit_rx) = reapable_grandchild.watch().unwrap();
```

Comment:
nit: we should use `?` instead of unwrap

Reply:
fixed.